src/Naneau/SemVer/Version.php

```diff
@@ -132,6 +133,93 @@ public function hasBuild()
     }
 
     /**
+     * Get the next logical version relative to the provided base version. If
+     * no base is supplied, base will be the same as the current version.
+     *
+     * @param  Version|string|null $base
+     * @return Version
+     */
+    public function next(/* $base */)
```

Comment:
It does indeed always return a new Version. It either returns with `$base->cleanCopy()` or `$next`, both new objects. Also, `$base` is never modified so there is no need for a clone in `ensureBase()`. (Just a note, as I'm sure you know, since `$preRelease` and `$build` are objects a simple clone wouldn't cut it. Would need to add a __clone() method to fix up those references.) 

As for `cleanCopy()` vs using clone, I went down the `__clone()` path but then I would need to unset `$preRelease` and `$build` after. There currently is no way to do that. I felt a distinct method would be clearer in this case regardless.